### PR TITLE
feat: lp: support multiple SP IDs

### PR DESCRIPTION
**Pull request by snadrus**
## Related Issues
Multi-Address support in Lotus-Provider will now offer different wallets per group of Miner IDs by expressing the same Address block as an array. A fallback singleton is provided as a migration path.

## Additional Info
[The issue for Multi-MinerID in Provider will be resolved
](https://github.com/filecoin-project/lotus/issues/11393)

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Comment by Stebalien** [comment]
I'm not sure if I have enough context to actually review this. There's nothing obviously wrong with the code.

File: cmd/lotus-provider/deps/deps.go [hunk 1]
```diff
@@ -236,18 +237,28 @@ Get it with: jq .PrivateKey ~/.lotus-miner/keystore/MF2XI2BNNJ3XILLQOJUXMYLUMU`,
...
-			addr, err := address.NewFromString(s)
-			if err != nil {
-				return err
+		for _, s := range deps.Cfg.Addresses {
+			for _, s := range s.MinerAddresses {
+				addr, err := address.NewFromString(s)
+				if err != nil {
+					return err
+				}
+				deps.Maddrs = append(deps.Maddrs, dtypes.MinerAddress(addr))
```
Comment (magik6k) [inline cmd/lotus-provider/deps/deps.go:246]: Should we check for duplicates and disallow them?

File: go.mod [hunk 2]
```diff
@@ -225,6 +231,7 @@ require (
 	github.com/hashicorp/golang-lru v0.6.0 // indirect
 	github.com/huin/goupnp v1.2.0 // indirect
 	github.com/iancoleman/orderedmap v0.1.0 // indirect
+	github.com/invopop/jsonschema v0.12.0
```
Comment (magik6k) [inline go.mod:234]: Afaict this is unused, go mod tidy should get rid of this.

File: storage/ctladdr/multiaddresses.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+type MultiAddressSelector struct {
```
Comment (magik6k) [inline storage/ctladdr/multiaddresses.go:14]: This is lotus-provider only right? In that case I'd move it to `provider/[address.go]`

**Review by magik6k — COMMENTED** [review]
Looks good, some small things to address.

Would be nice to do some manual testing with this if not done already

File: cmd/lotus-provider/deps/deps.go (hunk 1 above)
Comment (snadrus) [reply cmd/lotus-provider/deps/deps.go:246]: done

**Pull request merged**